processing/src/main/java/io/druid/segment/filter/SelectorFilter.java

```diff
@@ -36,26 +40,66 @@
 {
   private final String dimension;
   private final String value;
+  private final BinaryOperator operator;
 
   public SelectorFilter(
       String dimension,
-      String value
+      String value,
+      String operator
   )
   {
     this.dimension = dimension;
     this.value = value;
+    this.operator = BinaryOperator.get(operator);
+  }
+
+  public SelectorFilter(String dimension, String value)
+  {
+    this(dimension, value, null);
   }
 
   @Override
   public ImmutableBitmap getBitmapIndex(BitmapIndexSelector selector)
   {
-    return selector.getBitmapIndex(dimension, value);
+    final BitmapFactory factory = selector.getBitmapFactory();
+    final Indexed<String> values = selector.getDimensionValues(dimension);
+    switch (operator) {
+      case GT:
+      case GTE: {
+        int index = values.indexOf(value);
+        int start = index < 0 ? -index - 1 : operator == BinaryOperator.GT ? index + 1: index;
```

Comment:
some javadocs about this logic would make it easier to understand. 
Also, when the value is not present, there are two cases the
1) value is be greater than the current max which means no value match the filter
 2) value is less than the current min, i.e, all values are greater than value

does this needs to be fixed  or am i missing anything ?